texar/modules/connectors/connectors.py

```diff
@@ -591,30 +624,49 @@ class StochasticConnector(ConnectorBase):
             torch.Size.
             For example, to transform inputs to have decoder state size, set
             :python:`output_size=decoder.state_size`.
+        distribution: A instance of subclass of
+            :torch:`distributions.distribution.Distribution`,
+            Can be a class, its name or module path, or a class instance.
+        num_samples (optional): An ``int`` or ``int`` ``Tensor``.
+            Number of samples to generate. If not given,
+            generate a single sample. Note that if batch size has
+            already been included in :attr:`distribution`'s dimensionality,
+            :attr:`num_samples` should be left as ``None``.
         hparams (dict, optional): Hyperparameters. Missing
             hyperparamerter will be set to default values. See
             :meth:`default_hparams` for the hyperparameter sturcture and
             default values.
-        distribution: A instance of subclass of
-            :torch:`distributions.distribution.Distribution`,
-            Can be a class, its name or module path, or a class instance.
-        distribution_kwargs (dict, optional): Keyword arguments for the
-            distribution constructor. Ignored if :attr:`distribution` is a
-            class instance.
     """
 
     def __init__(self,
                  output_size: OutputSize,
-                 hparams: HParamsType = None,
-                 distribution: Union[Type[T], T, str] = 'MultivariateNormal',
-                 distribution_kwargs: Optional[Dict[str, Any]] = None,):
+                 distribution: T,
+                 num_samples: Optional[Union[int, torch.Tensor]] = None,
+                 hparams: Optional[HParamsType] = None):
         ConnectorBase.__init__(self, output_size, hparams)
```

Comment:
Could you change calls to the super class constructor to this:
```python
super().__init__(output_size, hparams)
```
This is the new convention in Python, and is less error-prone.